### PR TITLE
refactor sqlite3 pillar configuration as

### PR DESCRIPTION
**Pull request by akissa**
refactor sqlite3 pillar configuration as keys under pillar are overwritten by pillar data

### CODE REVIEW

File: salt/pillar/sqlite3.py [hunk 1]
```diff
@@ -78,12 +92,23 @@ def _get_options(self):
...
-            _sqlite3_opts = __opts__.get('pillar', {}).get('master', {})\
-                .get('pillar', {}).get('sqlite3')
-            if _sqlite3_opts is not None:
-                _opts = _sqlite3_opts
+        if 'sqlite3' in __opts__ and 'database' in __opts__['sqlite3']:
+            # new configuration
+            _opts = __opts__.get('sqlite3', {})
+        else:
+            # legacy configuration
+            salt.utils.warn_until(
```
Comment (cachedout) [inline salt/pillar/sqlite3.py]: Will need to import `salt.utils` first. :]

**Comment by cachedout** [comment]
@akissa Just one fix-up needed and then we can get this in. Thanks!

**Comment by akissa** [comment]
@cachedout Added the missing import.

**Comment by cachedout** [comment]
Thanks!

**Pull request merged**